app/models/manager_refresh/inventory_object_lazy.rb

```diff
@@ -36,7 +36,7 @@ def transitive_dependency?
       # and a :stack is a relation to another object, in the InventoryObject object,
       # then this relation is considered transitive.
       !!(key && (inventory_collection.dependency_attributes.keys.include?(key) ||
-        inventory_collection.model_class.reflect_on_association(key)))
+        (inventory_collection.model_class && inventory_collection.model_class.reflect_on_association(key))))
```

Comment:
Prefer `.try`

```ruby
inventory_collection.model_class.try(:reflect_on_association, key)
```

Also this conditional is impossible to understand, and the comment doesn't make it any easier.  Can you explain what it's actually doing?  Perhaps we can come up with a better comment if described another way,